### PR TITLE
OSDOCS 7180- Allow Ingress to Modify the HAProxy Log Length when using a Sidecar

### PR DESCRIPTION
**Pull request by jmanthei**
Version(s): 4.14

Issue: (https://issues.redhat.com/browse/OSDOCS-7180)

Link to docs preview: https://65658--docspreview.netlify.app/openshift-enterprise/latest/networking/ingress-operator#nw-configure-ingress-access-logging_configuring-ingress 

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by jmanthei** [comment]
@Miciah or other NE folks, what should go in "sample code specifics here" at the end of the section?

**Comment by jmanthei** [comment]
@ShudiLi please QE check

File: modules/nw-configure-ingress-access-logging.adoc [hunk 1]
```diff
@@ -116,3 +116,47 @@ spec:
...
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  replicas: 2
+  logging:
+    access:
+      destination:
+        type: Container
+        container:
+          maxLength: 4096
```
Comment (ShudiLi) [inline modules/nw-configure-ingress-access-logging.adoc:160]: maxLength here should be 8192, thanks.

**Comment by ShudiLi** [comment]
except the maxLength for the Container sidecar, others look good to me.

File: modules/nw-configure-ingress-access-logging.adoc (hunk 1 above)
Comment (jmanthei) [reply modules/nw-configure-ingress-access-logging.adoc]: @ShudiLi is that for both examples?

Comment (ShudiLi) [reply modules/nw-configure-ingress-access-logging.adoc]: @jmanthei  Sorry for the late reply. Just for the Container, for the syslog, it is still 4096.

**Comment by ShudiLi** [comment]
/lgtm
thanks

File: modules/nw-configure-ingress-access-logging.adoc [hunk 2]
```diff
@@ -116,3 +116,47 @@ spec:
   logging:
     access: null
 ----
+
+Allowing IngressController to Modify the HAProxy Log Length when using a Sidecar
+
+* Configure `httpLogFormat` within IngressController. Leaving it blank will default to standard http log format. 
+Use `spec.logging.access.destination.syslog.maxLength` if you're using `spec.logging.access.destination.type: Syslog`.
```
Comment (jeana-redhat) [inline modules/nw-configure-ingress-access-logging.adoc:123]: A couple points:
1. This says to configure the `httpLogFormat` parameter, but does not show that in the YAML. Should it be included?
2. Either "the `IngressController` resource" (labelled resource name) or "the Ingress Controller" (plain text)
3. Generally avoid contractions, so sub "you are" in for "you're" (here and in the line later on)

**Review by jeana-redhat — COMMENTED** [review]
Just a few things to think about here. 

Waaaaaay out of scope for this effort, but I do want to note that this module should probably be reformatted to match our usual patterns. Makes sense that you matched what already exists for this addition :)

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

File: modules/nw-configure-ingress-access-logging.adoc [hunk 3]
```diff
@@ -116,3 +116,47 @@ spec:
   logging:
     access: null
 ----
+
+Allowing IngressController to Modify the HAProxy Log Length when using a Sidecar
```
[the remark this answers is not in the document]
Comment (jmanthei) [reply modules/nw-configure-ingress-access-logging.adoc:120]: ty

File: modules/nw-configure-ingress-access-logging.adoc (hunk 2 above)
Comment (Miciah) [reply modules/nw-configure-ingress-access-logging.adoc]: I think the two sentences, "Configure `httpLogFormat` within IngressController. Leaving it blank will default to standard http log format. " can be omitted from the list item about configuring maximum log length.  `httpLogFormat` and `maxLength` are separate API fields.

**Comment by jmanthei** [comment]
/label merge-review-needed

Commits just squashed.

File: modules/nw-configure-ingress-access-logging.adoc [hunk 4]
```diff
@@ -116,3 +116,46 @@ spec:
   logging:
     access: null
 ----
+
+Allowing IngressController to modify the HAProxy log length when using a sidecar.
+
+Use `spec.logging.access.destination.syslog.maxLength` if you are using `spec.logging.access.destination.type: Syslog`.
+
++
```
Comment (abhatt-rh) [inline modules/nw-configure-ingress-access-logging.adoc:124]: @jmanthei If you are following the formatting as the rest of the module, consider using the unordered list (`*`) (highlighted in green). If  you are not using a list, then you may want to remove the list continuation (`+`) (highlighted in orange) 
image

File: modules/nw-configure-ingress-access-logging.adoc [hunk 5]
```diff
@@ -116,3 +116,46 @@ spec:
...
+    access:
+      destination:
+        type: Syslog
+        syslog:
+          address: 1.2.3.4
+          maxLength: 4096
+          port: 10514
+----
+Use `spec.logging.access.destination.container.maxLength` if you are using `spec.logging.access.destination.type: Container`.
++
```
Comment (abhatt-rh) [inline modules/nw-configure-ingress-access-logging.adoc:144]: Same comment as above

File: modules/nw-configure-ingress-access-logging.adoc [hunk 6]
```diff
@@ -116,3 +116,46 @@ spec:
   logging:
     access: null
 ----
+
+Allowing IngressController to modify the HAProxy log length when using a sidecar.
```
Comment (abhatt-rh) [inline modules/nw-configure-ingress-access-logging.adoc:120]: Sorry, but don't see the peer review feedback [here](https://github.com/openshift/openshift-docs/pull/65658#discussion_r1371799047) impemented. Kindly, considering implementing as is makes the instructions parallel with the rest and suggests correcting the object reference. 
You may use either a [general reference](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-references) or use an [object reference](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references) for object names.

**Review by abhatt-rh — COMMENTED** [review]
Hi @jmanthei, I came across a few formatting issues during my merge review. PTAL and let me know when you are ready for a merge.

File: modules/nw-configure-ingress-access-logging.adoc (hunk 6 above)
Comment (jmanthei) [reply modules/nw-configure-ingress-access-logging.adoc]: @abhatt-rh All comments incorporated, will squash.

**Pull request merged**